src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -79,13 +79,16 @@ public static int NeedsEscaping(ReadOnlySpan<byte> value, JavaScriptEncoder enco
             return idx;
         }
 
-        public static int NeedsEscaping(ReadOnlySpan<char> value, JavaScriptEncoder encoder)
+        public static unsafe int NeedsEscaping(ReadOnlySpan<char> value, JavaScriptEncoder encoder)
         {
             int idx;
 
             if (encoder != null)
             {
-                idx = encoder.FindFirstCharacterToEncodeUtf8(MemoryMarshal.Cast<char, byte>(value));
+                fixed (char* ptr = value)
+                {
+                    idx = encoder.FindFirstCharacterToEncode(ptr, value.Length);
+                }
                 goto Return;
```

Comment:
For v5 we may want to add a overload of `FindFirstCharacterToEncode(ReadOnlySpan<char)` to S.T.Encoding.Web so consumers don't have to use unsafe pinning code.